tensorflow_serving/batching/batching_util.h

```diff
@@ -0,0 +1,206 @@
+  PaddingResult res;
+  switch (num_dims) {
+    case 0: {
+      std::array<std::pair<int32, int32>, 0> padding;
+      padding = CreatePadding<0>(tensor, max_dim_sizes);
+      PadTensor<T, 0> padding_functor = PadTensor<T, 0>();
+      res = padding_functor(tensor, padding);
+      break;
+    }
+    case 1: {
+      std::array<std::pair<int32, int32>, 1> padding;
+      padding = CreatePadding<1>(tensor, max_dim_sizes);
+      PadTensor<T, 1> padding_functor = PadTensor<T, 1>();
+      res = padding_functor(tensor, padding);
+      break;
+    }
+    case 2: {
+      std::array<std::pair<int32, int32>, 2> padding;
+      padding = CreatePadding<2>(tensor, max_dim_sizes);
+      PadTensor<T, 2> padding_functor = PadTensor<T, 2>();
+      res = padding_functor(tensor, padding);
+      break;
+    }
+    case 3: {
+      std::array<std::pair<int32, int32>, 3> padding;
+      padding = CreatePadding<3>(tensor, max_dim_sizes);
+      PadTensor<T, 3> padding_functor = PadTensor<T, 3>();
+      res = padding_functor(tensor, padding);
+      break;
+    }
+    case 4: {
+      std::array<std::pair<int32, int32>, 4> padding;
+      padding = CreatePadding<4>(tensor, max_dim_sizes);
+      PadTensor<T, 4> padding_functor = PadTensor<T, 4>();
+      res = padding_functor(tensor, padding);
+      break;
+    }
+    case 5: {
+      std::array<std::pair<int32, int32>, 5> padding;
+      padding = CreatePadding<5>(tensor, max_dim_sizes);
+      PadTensor<T, 5> padding_functor = PadTensor<T, 5>();
+      res = padding_functor(tensor, padding);
+      break;
+    }
+    case 6: {
+      std::array<std::pair<int32, int32>, 6> padding;
+      padding = CreatePadding<6>(tensor, max_dim_sizes);
+      PadTensor<T, 6> padding_functor = PadTensor<T, 6>();
+      res = padding_functor(tensor, padding);
+      break;
+    }
+    default:
+    // only ranks from 0 to 6 are supported
+    // (like in tensorflow/core/kernels/pad_op.cc)
+      res = {Tensor(), errors::InvalidArgument(
+            "Only tensors with rank from 0 to 6 can be padded.")};
+  }
+  return res;
+}
+
+// For batch of inputs calculates maximum dim sizes across all tensors
+// with the same name.
+// These dim sizes are used later to calculate padding amount for each tensor.
+// For zeroth dimension max_dim_size is set to zero,
+// because padding is not performed in this dimension.
+// For example, for batch containing three tasks with the following inputs
+// (instead of tensors there are their shapes):
+//
+// task1: {'features': [100, 500, 300], 'true_labels': [100]}
+// task2: {'features': [100, 200, 123], 'true_labels': [100]}
+// task3: {'features': [200, 100, 400], 'true_labels': [200]}
+//
+// the following map will be generated:
+// {'features': [0, 500, 400], 'true_labels': [0]}
+std::map<string, std::vector<int>> CalculateMaxDimSizes(
+    const std::vector<std::vector<std::pair<string, Tensor>>>& batch);
+
+// Pads tensor so that its shape becomes as specified in max_dim_sizes,
```

Comment:
Please document what values are used for padding (see my other comment for discussion of what we want to do in terms of zeros or existing values).